shangguanwang/NFT-marketplace
# Connect Wallet using RainbowKit

## Description
Replace the "Sign up" dummy button with a "Connect Wallet" button

## Acceptance Criteria
- [x] install RainbowKit and its related dependencies
- [x] obtain a projectId from [WalletConnect Cloud](https://cloud.walletconnect.com/) and an API key from [Alchemy](https://www.alchemy.com/) to configure the wallet setting
- [x] customize the button text and color 

## Related Issue
closes #26 

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|  ✓   | :link: Update dependencies |
|   ✓  | :scroll: Docs              |

## Updates
### Before
![button-before](https://github.com/shangguanwang/NFT-marketplace/assets/24996005/b8e83cd7-2e15-4c6d-b541-67e356e13aba)

### After
![button-after](https://github.com/shangguanwang/NFT-marketplace/assets/24996005/4b36d5c8-fdc1-4b3e-b685-044c711e24bf)
Once clicked, you should see the modal
![connect wallet modal](https://github.com/shangguanwang/NFT-marketplace/assets/24996005/9e1e5cf4-5b2e-4c9c-9ecb-a2a9ddec3c95)

## Testing Steps / QA Criteria
- From your terminal, pull down this branch with `git pull origin feat-connect-wallet` and check that branch out with `git checkout feat-connect-wallet`
- Run `npm run dev` to launch the app
- Click on the `Connect Wallet` button if you already have a Metamask wallet, after successfully connected you should see your balance
